### PR TITLE
feat: add 'open in new tab' option

### PR DESCRIPTION
**Pull request by cbicuti-godaddy**
RSV-1712 - New Tab Option for Cart and Domain

### CODE REVIEW

**Review by cberesford-godaddy — CHANGES_REQUESTED** [review]
Can you please add unit tests for this new functionality?
Something along the lines of "It should apply target=_blank when button_new_tab is checked"

File: .dev/src/blocks/product/components/inspector.js [hunk 1]
```diff
@@ -65,6 +65,13 @@ const Inspector = ( { posts, media, attributes, setAttributes } ) => {
 					value={ attributes.button_label }
 					onChange={ ( buttonLabel ) => setAttributes( { button_label: buttonLabel } ) }
 				/>
+				<CheckboxControl
+					label={ __( 'Open in New Tab', 'reseller-store' ) }
```
Comment (cburton-godaddy) [inline .dev/src/blocks/product/components/inspector.js:69]: **nitpick (non-blocking):** Casing does not match the other description for this.

`Display results in a new tab` vs `Open in New Tab`

Also latter misses the `a` too. Make these as consistent as possible whilst still making sense in context.

File: includes/modules/rstore-fl-product.php [hunk 2]
```diff
@@ -84,6 +84,15 @@ public function __construct() {
 							'description' => __( 'Leave blank to hide button', 'reseller-store' ),
 							'default'     => __( 'Add to cart', 'reseller-store' ),
 						),
+						'button_new_tab'     => array(
+							'type'    => 'select',
+							'label'   => __( 'Open in new tab', 'reseller-store' ),
```
Comment (cburton-godaddy) [inline includes/modules/rstore-fl-product.php:89]: **nitpick (non-blocking):** More inconsistency, casing is different to the other `Open in New Tab`

File: includes/modules/rstore-vc-product.php [hunk 3]
```diff
@@ -86,6 +86,18 @@ public function vc_mapping() {
 						'group'       => __( 'Display', 'reseller-store' ),
 					),
 
+					array(
+						'type'       => 'checkbox',
+						'holder'     => 'div',
+						'class'      => 'text-class',
+						'heading'    => __( 'Open in new tab', 'reseller-store' ),
```
Comment (cburton-godaddy) [inline includes/modules/rstore-vc-product.php:93]: **nitpick (non-blocking):** Make consistent

File: includes/widgets/class-product.php [hunk 4]
```diff
@@ -179,7 +179,7 @@ public function widget( $args, $instance ) {
 
 			if ( ! empty( $data['button_label'] ) ) {
 
-				$content .= rstore_add_to_cart_form( $post_id, false, $data['button_label'], $data['text_cart'], $data['redirect'] );
+				$content .= rstore_add_to_cart_form( $post_id, false, $data['button_label'] . 'a', $data['button_new_tab'], $data['text_cart'], $data['redirect'] );
```
Comment (cburton-godaddy) [inline includes/widgets/class-product.php:182]: **question (non-blocking):** What's the `. 'a'` for here?

**Review by cburton-godaddy — CHANGES_REQUESTED** [review]
+1 on tests too

**Review by cburton-godaddy — APPROVED** [review]

**Pull request merged**